### PR TITLE
Using enhanced loops where possible

### PR DESCRIPTION
**Pull request by dmatej**
- Cleanup made by Eclipse IDE and revisited manually

### CODE REVIEW

File: src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractEncryptOutputProcessor.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public void init(OutputProcessorChain outputProcessorChain) throws XMLSecurityEx
             String encryptionSymAlgorithm = securityProperties.getEncryptionSymAlgorithm();
             try {
                 //initialize the cipher
-                String jceAlgorithm = JCEAlgorithmMapper.translateURItoJCEID(encryptionSymAlgorithm);
+                String jceAlgorithm = JCEMapper.translateURItoJCEID(encryptionSymAlgorithm);
```
Comment (coheigea) [inline src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractEncryptOutputProcessor.java:163]: Was this change intended?

File: src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractSignatureOutputProcessor.java [hunk 2]
```diff
@@ -181,8 +182,8 @@ protected void setActiveInternalSignatureOutputProcessor(
     protected DigestOutputStream createMessageDigestOutputStream(String digestAlgorithm)
             throws XMLSecurityException {
 
-        String jceName = JCEAlgorithmMapper.translateURItoJCEID(digestAlgorithm);
-        String jceProvider = JCEAlgorithmMapper.getJCEProviderFromURI(digestAlgorithm);
+        String jceName = JCEMapper.translateURItoJCEID(digestAlgorithm);
+        String jceProvider = JCEMapper.getJCEProviderFromURI(digestAlgorithm);
```
Comment (coheigea) [inline src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractSignatureOutputProcessor.java:186]: Why this change?

File: src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignedInfo.java [hunk 3]
```diff
@@ -100,17 +100,13 @@ public DOMSignedInfo(CanonicalizationMethod cm, SignatureMethod sm,
...
-                "contain at least one entry");
+            throw new IllegalArgumentException("list of references must contain at least one entry");
         }
-        for (int i = 0, size = this.references.size(); i < size; i++) {
-            Object obj = this.references.get(i);
-            if (!(obj instanceof Reference)) {
-                throw new ClassCastException("list of references contains " +
-                    "an illegal type");
+        for (Object obj : this.references) {
+            if (!Reference.class.isInstance(obj)) {
```
Comment (coheigea) [inline src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignedInfo.java:108]: According to https://rules.sonarsource.com/java/RSPEC-6202, the isInstance operator is preferred when the class name is known at compile time.

**Review by coheigea — COMMENTED** [review]
Also 2 PMD failures:

[INFO] PMD Failure: org.apache.xml.security.stax.impl.processor.output.AbstractEncryptOutputProcessor:104 Rule:CollapsibleIfStatements Priority:3 These nested if statements could be combined.
[INFO] PMD Failure: org.apache.xml.security.stax.impl.processor.output.AbstractSignatureOutputProcessor:156 Rule:CollapsibleIfStatements Priority:3 These nested if statements could be combined.

File: src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignedInfo.java (hunk 3 above)
Comment (dmatej) [reply src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignedInfo.java:108]: This is a bit complicated, first I removed the for loop as it should not be possible to have an incompatible element IF user would keep rules (generics). But there is an unit test which proves that it is possible to pass incompatible type. Maybe it should be refactored later - at least I will comply with rules and bring back the `instanceof`.

File: src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractSignatureOutputProcessor.java (hunk 2 above)
Comment (dmatej) [reply src/main/java/org/apache/xml/security/stax/impl/processor/output/AbstractSignatureOutputProcessor.java:186]: Automatic change made by Eclipse - the method is defined in the parent class, but it is static. There is no such method in `JCEAlgorithMapper`. In fact the `JCEAlgorithmMapper` should NOT inherit from `JCEMapper`, both contain just static methods and both should be final. But that would be a breaking change, however I don't think anyone "extended" these classes.

I would keep it there as it is here, because it is the right direction. In "sanct4" branch we can continue with breaking changes.

**Review by coheigea — APPROVED** [review]
